app/controllers/impressionist_controller.rb

```diff
@@ -149,13 +149,13 @@ def session_hash
       # # request.session_options[:id].encode("ISO-8859-1")
       if Rails::VERSION::MAJOR >= 4
         session["init"] = true
-        id = session.id.to_s
+        id = session.id.public_id.to_s
```

Comment:
@rzjfr 
If `session.id` is null, It will raise exception here. You should check null value for `session.id`

Reply:
correct, title is misleading

Reply:
Thanks digging into this @rzjfr. I think something like this might work for any rack version `>= 2.1.0`

```ruby
def session_hash
  if Rails::VERSION::MAJOR >= 4
    session['init'] = true
    id = session.id
  else
    id = request.session_options[:id]
  end

  if id.respond_to?(:cookie_value)
    id = id.cookie_value
  elsif id.respond_to?(:public_id)
    id = id.public_id
  end

  id
end
```